### PR TITLE
Use cluster api map function

### LINKED ISSUE #146: Use cluster API machineToInfrastructureMachine
/kind bug

**What steps did you take and what happened:**
This got lost in a merge commit, basically replace the mapfn that maps a machine to an infrastructure machine to the one provided by cluster-api

/cc @ericzhang08

### PR DESCRIPTION
**Pull request by ericzhang08**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #146 

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: controllers/dockermachine_controller.go [hunk 1]
```diff
@@ -65,32 +63,8 @@ func (r *DockerMachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		Watches(
 			&source.Kind{Type: &infrastructurev1alpha2.DockerMachine{}},
```
Comment (ncdc) [inline controllers/dockermachine_controller.go]: This needs to be watching a cluster-api Machine

Comment (ericzhang08) [reply controllers/dockermachine_controller.go]: Thank you. I have fixed it.

File: controllers/dockermachine_controller.go [hunk 2]
```diff
@@ -63,34 +61,10 @@ func (r *DockerMachineReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&infrastructurev1alpha2.DockerMachine{}).
 		Watches(
-			&source.Kind{Type: &infrastructurev1alpha2.DockerMachine{}},
+			&source.Kind{Type: &capiv1alpha2.Machine{}},
 			&handler.EnqueueRequestsFromMapFunc{
-				ToRequests: MachineToDockerMachine(capiv1alpha2.SchemeGroupVersion.WithKind("Machine")),
+				ToRequests: util.MachineToInfrastructureMapFunc(capiv1alpha2.SchemeGroupVersion.WithKind("Machine")),
```
Comment (detiber) [inline controllers/dockermachine_controller.go]:
```suggestion
				ToRequests: util.MachineToInfrastructureMapFunc(infrastructurev1alpha2.SchemeGroupVersion.WithKind("DockerMachine")),
```

Comment (ncdc) [reply controllers/dockermachine_controller.go]: Good catch!

**Comment by ncdc** [comment]
LGTM. Would you mind squashing your commits?

**Comment by chuckha** [comment]
/hold 

whoops, sorry andy, came from my email, please clear the hold whenever you like

**Comment by ncdc** [comment]
@chuckha no worries, it already had a hold 😄

**Comment by chuckha** [comment]
/hold

oops, so sorry @ericzhang08 , can you squash your commits?

**Pull request merged**